app/deepseg.cc

```diff
@@ -613,11 +627,27 @@ int main(int argc, char* argv[]) try {
 		loopback_free(lbfd);
 	});
 
-
 	// Processing components, all at capture true geometry
-	cv::Mat mask(capGeo.value().second, capGeo.value().first, CV_8U);
+	cv::Mat mask;
+	if ( crop_region.height ) {
+		cv::Mat masktmp((int)(crop_region.height), (int)(crop_region.width), CV_8U);
+		mask = masktmp;
+	} else {
+		cv::Mat masktmp(capGeo.value().second, capGeo.value().first, CV_8U);
+		mask = masktmp;
+	}
```

Comment:
```suggestion
	if (crop_region.height) {
		mask = cv::Mat((int)crop_region.height, (int)crop_region.width, CV_8U);
	} else {
		mask = cv::Mat(capGeo->second, capGeo->first, CV_8U);
	}
```

Reply:
This is modified line 633 as:

```
	std::optional<std::pair<size_t, size_t>> mask_dim = capGeo;
	if ( crop_region.height ) {
		mask_dim = {crop_region.width, crop_region.height};
	}
	cv::Mat mask(mask_dim.value().second, mask_dim.value().first, CV_8U);``
```
within my cropping branch

Reply:
Note the `.value().first` -> `->first` change in my suggestion. Saves quite a bit of typing.

Reply:
I had a look at the already provided code and used the same syntax. I had not looked correctly to your suggestion. I will try this.